compiler/injectdestructors.nim

```diff
@@ -277,7 +277,8 @@ proc deepAliases(dest, ri: PNode): bool =
 proc genSink(c: var Con; s: var Scope; dest, ri: PNode; flags: set[MoveOrCopyFlag] = {}): PNode =
   if (c.inLoopCond == 0 and (isUnpackedTuple(dest) or IsDecl in flags or
       (isAnalysableFieldAccess(dest, c.owner) and isFirstWrite(dest, c)))) or
-      isNoInit(dest) or IsReturn in flags:
+      isNoInit(dest) or IsReturn in flags or
+      tfByRef in dest.typ.flags:
```

Comment:
I think the logic should be "it's .byref and also has its =sink disabled".

Reply:
make sense

Reply:
This is still not correct as it can introduce leaks for code like `var x = create(); x = create()`. The idea here is that `.byref` types do not have any bitwise copies. We need to rethink this. It probably means the whole idea of `=sink() .error` is just wrong. `=sink` is fine and doesn't copy and shallow copies are prevented by `.byref`.